node/silkworm/common/log.hpp

```diff
@@ -17,60 +17,79 @@
 #ifndef SILKWORM_COMMON_LOG_HPP_
 #define SILKWORM_COMMON_LOG_HPP_
 
-#include <mutex>
-
-#include <silkworm/common/tee.hpp>
-
-namespace silkworm {
-
-// stream labeled logging output - e.g.
-//	  SILKWORM_LOG(LogInfo) << "All your " << num_bases << " base are belong to us\n";
-//
-#define SILKWORM_LOG(level_)                   \
-    if ((level_) < silkworm::log_verbosity_) { \
-    } else                                     \
-        log_(level_) << " "
-
-// change the logging verbosity level - default level is LogInfo
-//
-#define SILKWORM_LOG_VERBOSITY(level_) (silkworm::log_verbosity_ = (level_))
-
-// change if thread is logged (true) or not (false) - default is false
-//
-#define SILKWORM_LOG_THREAD(log_thread_) (silkworm::log_thread_enabled_ = (log_thread_))
-
-// available verbosity levels
-enum class LogLevel { Trace, Debug, Info, Warn, Error, Critical, None };
-
-// change the logging output streams - default is (cerr, null_stream())
-//
-#define SILKWORM_LOG_STREAMS(stream1_, stream2_) silkworm::log_set_streams_((stream1_), (stream2_));
-
-// silence
-std::ostream& null_stream();
-
-//
-// Below are for access via macros ONLY.
-// Placing them in detail namespace prevents use of macros in nested namespaces of silkworm :(
-//
-extern LogLevel log_verbosity_;
-extern bool log_thread_enabled_;
-void log_set_streams_(std::ostream& o1, std::ostream& o2);
-class log_ {
+#include <filesystem>
+#include <sstream>
+
+#include <silkworm/common/terminal.hpp>
+
+namespace silkworm::log {
+
+//! \brief Available verbosity levels
+enum class Level {
+    None,      // Simple logging line with no severity (eg. build info)
+    Critical,  // An error there's no way we can recover from
+    Error,     // We encountered an error which we might be able to recover from
+    Warn,      // Something happened and user might have the possibility to amend the situation
+    Info,      // Info messages on regular operations
+    Debug,     // Debug information
+    Trace      // Trace calls to functions
+};
+
+//! \brief Holds logging configuration
+struct Settings {
+    bool log_std_out{false};    // Whether console logging goes to std::cout or std::cerr (default)
+    bool log_utc{false};        // Whether timestamps should be in UTC or imbue local timezone
+    bool log_nocolor{false};    // Whether to disable colorized output
+    bool log_threads{false};    // Whether to print thread ids in log lines
+    unsigned log_verbosity{4};  // Log verbosity level
```

Comment:
Please use the enum value instead of the magic constant.

Reply:
Done in 0186377